src/trace/tracer.ts

```diff
@@ -147,24 +154,24 @@ class ContextTracer implements SnapshotterDelegate, ActionListener {
     return (page as any)[pageIdSymbol];
   }
 
-  async onActionCheckpoint(name: string, metadata: ActionMetadata): Promise<void> {
+  async onActionCheckpoint(name: string, sdkObject: SdkObject, metadata: CallMetadata): Promise<void> {
     const snapshotId = createGuid();
     snapshotsForMetadata(metadata).push({ name, snapshotId });
-    await this._snapshotter.forceSnapshot(metadata.page, snapshotId);
+    await this._snapshotter.forceSnapshot(sdkObject.attribution.page!, snapshotId);
   }
 
-  async onAfterAction(result: ProgressResult, metadata: ActionMetadata): Promise<void> {
+  async onAfterAction(result: ActionResult, sdkObject: SdkObject, metadata: CallMetadata): Promise<void> {
     const event: trace.ActionTraceEvent = {
       timestamp: monotonicTime(),
       type: 'action',
       contextId: this._contextId,
-      pageId: this.pageId(metadata.page),
-      action: metadata.type,
-      selector: typeof metadata.target === 'string' ? metadata.target : undefined,
-      value: metadata.value,
+      pageId: this.pageId(sdkObject.attribution.page!),
+      objectType: metadata.type,
+      method: metadata.method,
+      params: metadata.params,
```

Comment:
Writing all parameters into the json could be expensive (think all evaluation scripts), and also easily non-serializable. We should follow up.